rpcclient/infrastructure.go

```diff
@@ -924,13 +931,60 @@ func (c *Client) nextPostRequest() chan *jsonRequest {
 // while allowing the sender to continue running asynchronously.  It must be run
 // as a goroutine.
 func (c *Client) sendPostHandler() {
+	// Add an errgroup to handle the post messages in goroutines.
+	var eg errgroup.Group
+
+	// Limit the number of concurrent RPC calls.
+	//
+	// TODO(yy): Ideally we should pass the `-rpcthreads` used in
+	// `bitcoind` here to better increase the performance.
+	const numThreads = 4
+	eg.SetLimit(numThreads)
+
+	// lowReqCounter counts the number of low priority requests.
+	lowReqCounter := 0
```

Comment:
Should we use an explicit `uint64` type here to avoid rolling over into negative? Wouldn't really matter too much as we're only using the modulus, but would give us some more explicit behavior.

Reply:
updated